docs: add terminology (#6154)
scalableminds/webknossos: docs/terminology.md

# Terminology

> ## Dataset

Reviewer: I would use plural here as well:
```suggestion
## Datasets
```

A **dataset**, can consist of multiple **layer**s, which may be

* **segmentation layer**s or
* **color layer**s.

Layers contain image data in one or multiple **magnification**s or short **mag**s (see [mipmap](https://en.wikipedia.org/wiki/Mipmap) or [image pyramids](https://en.wikipedia.org/wiki/Pyramid_(image_processing)) for similar concepts.)

> The magnification `4` or `4-4-4` describe a downsampling factor of 4 in each dimension, `4-4-2` specifies anisotropic downsampling.

Reviewer 2: ```suggestion
The magnification `4` or `4-4-4` describes a downsampling factor of 4 in each dimension, `4-4-2` specifies anisotropic downsampling.
```

> The image data in full resolution is referred to as the **finest** mag, downsampled variants are more **coarse**.

Reviewer: ```suggestion
The image data in full resolution is referred to as the **finest** mag, e.g. `1-1-1`, downsampled variants are more **coarse**. 
```

The **voxel size** describes the size of a voxel in mag `1`, the default unit is *nm* if not specified otherwise.

The underlying file formats zarr and wkw use **chunks** as a compressible unit and **shards** as a storage unit. A wkw block corresponds to a chunk, and a file to a shard.

## Annotation

An **annotation** can consist of one or multiple **annotation layers**, which can either be a

> * **volume annotation layer**, or **volume layer** short ([main guide](./volume_annotation.md)), or

Reviewer: Just `short` sounds off. Is it `in short`? Or `*volume annotation layer* (shorthand *volume layer* ... `

```suggestion
* **volume annotation layer**, or **volume layer** in short ([main guide](./volume_annotation.md)), or
```

> * **skeleton annotation layer**, or **skeleton layer** and **skeleton** short, ([main guide](./skeleton_annotation.md)).

Reviewer: same here

Author: ```suggestion
* **skeleton annotation layer**, or **skeleton layer** and **skeleton** in short, ([main guide](./skeleton_annotation.md)).
```

**Volume-only** and **skeleton-only** annotations are restricted to the specific annotation layer type.

A **skeleton** consists of **trees** with nodes and edges, which may be organized in **groups**.
A tree is supposed to be a single connected component. Despite the name, it may contain cycles.

**Branchpoint**s are boolean markers on nodes, that can be jumped to easily in the frontend. They are not tied to any network-specific properties of the node.

## Tasks & Projects

> A **task** specifies a desired annotation result. A **task instance** refers to the concrete annotation on a task, which can be done redundantly for a task. Task instances may also be abbreviated as tasks if the context is clear.

Reviewer: > A **task instance** refers to the concrete annotation on a task, which can be done redundantly for a task.

I am not sure this is correct. A task instance is an instance of a task :-D it holds the metadata about how much time was used to work on the task, who is working on the task, reference to the task type, the ID/url of the respective annotation ...

The linked `docs` side describes them as:
```
Task: Is an assignment for a small piece of work. A Team Manager or Admin creates Tasks with information about the Task Type, the referenced dataset, starting positions, and advanced options. These tasks will be distributed to eligible users.

Task Instance: Some Tasks need to be completed redundantly by multiple users to assure quality. The single assignments of the tasks are called Instances.
```

I probably should be something like this:
```
A **task instance** refers to the concrete copy of a task assigned to a user for annotation work. A task may be handed out to several users for redundancy resulting in several task instances, each containing their respective annotation.
```

Author: ```suggestion
A **task** specifies a desired annotation result. A **task instance** refers to the concrete copy of a task assigned to a user for annotation work. A task may be handed out to several users for redundancy resulting in several task instances, each containing their respective annotation. Task instances may also be abbreviated as tasks if the context is clear.
```
